devicetypes/sky-nie/min-smart-plug.src/min-smart-plug.groovy

```diff
@@ -0,0 +1,530 @@
+        cmds << switchBinaryGetCmd()
+    }
+
+    if (state.resyncAll || !device.currentValue("firmwareVersion")) {
+        cmds << versionGetCmd()
+    }
+
+    configParams.each { param ->
+        def storedVal = getParamStoredValue(param.num)
+        def paramVal = param.value
+
+        if (state.resyncAll || ("${storedVal}" != "${paramVal}")) {
+            logDebug "Changing ${param.name}(#${param.num}) from ${storedVal} to ${paramVal}"
+            cmds << configSetCmd(param, paramVal)
+            cmds << configGetCmd(param)
+        }
+    }
+
+    state.resyncAll = false
+    if (cmds) {
+        sendCommands(delayBetween(cmds, 500))
+    }
+    return []
+}
+
+def ping() {
+    logDebug "ping()..."
+
+    return [ switchBinaryGetCmd() ]
+}
+
+def on() {
+    logDebug "on()..."
+
+    return [ switchBinarySetCmd(0xFF) ]
+}
+
+def off() {
+    logDebug "off()..."
+
+    return [ switchBinarySetCmd(0x00) ]
+}
+
+def refresh() {
+    logDebug "refresh()..."
+
+    refreshSyncStatus()
+
+    sendCommands([switchBinaryGetCmd()])
+}
+
+private sendCommands(cmds) {
+    if (cmds) {
+        def actions = []
+        cmds.each {
+            actions << new physicalgraph.device.HubAction(it)
+        }
+        sendHubCommand(actions)
+    }
+    return []
+}
+
+private versionGetCmd() {
+    return secureCmd(zwave.versionV1.versionGet())
+}
+
+private switchBinaryGetCmd() {
+    return secureCmd(zwave.switchBinaryV1.switchBinaryGet())
+}
+
+private switchBinarySetCmd(val) {
+    return secureCmd(zwave.switchBinaryV1.switchBinarySet(switchValue: val))
+}
+
+private configSetCmd(param, value) {
+    return secureCmd(zwave.configurationV1.configurationSet(parameterNumber: param.num, size: param.size, scaledConfigurationValue: value))
+}
+
+private configGetCmd(param) {
+    return secureCmd(zwave.configurationV1.configurationGet(parameterNumber: param.num))
+}
+
+private secureCmd(cmd) {
+    try {
+        if (zwaveInfo?.zw?.contains("s") || ("0x98" in device?.rawDescription?.split(" "))) {
+            return zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+        }
+        else {
+            return cmd.format()
+        }
+    }
+    catch (ex) {
+        return cmd.format()
+    }
+}
+
+def parse(String description) {
+    def result = []
+    try {
+        def cmd = zwave.parse(description, commandClassVersions)
+        if (cmd) {
+            result += zwaveEvent(cmd)
+        }
+        else {
+            log.warn "Unable to parse: $description"
+        }
+
+        updateLastCheckIn()
+    }
+    catch (e) {
+        log.error "${e}"
+    }
+    return result
+}
+
+private updateLastCheckIn() {
+    if (!isDuplicateCommand(state.lastCheckInTime, 60000)) {
+        state.lastCheckInTime = new Date().time
+
+        def evt = [name: "lastCheckIn", value: convertToLocalTimeString(new Date()), displayed: false]
+
+        sendEvent(evt)
+
+        if (childDevices) {
```

Comment:
You refer to childDevices couple of times in the code, but you don't create any. May I ask why?

Reply:
OK, I remove  all about childDevices, they sync from other file, but not used.